### ISSUE TITLE
`starting readStream` error log

### ISSUE THREAD

**Issue by sir4ur0n** [launchdarkly/haskell-server-sdk#62]
**Describe the bug**
At application startup (and maybe other times too, I haven't checked), we get this annoying error log in all our environments, which triggers our Grafana alerting:
```
starting readStream
```
This log comes from https://github.com/launchdarkly/haskell-server-sdk/blob/main/src/LaunchDarkly/Server/Network/Streaming.hs#L193

**To reproduce**
Launch an application using LaunchDarkly Haskell library.

**Expected behavior**
Either the log is at a normal level (DEBUG or INFO), or the message should better explain what the problem actually is.

**Logs**
```
{"at":"2023-09-26T06:56:18.692576032Z","loc":{"loc_col":6,"loc_fn":"src/LaunchDarkly/Server/Network/Streaming.hs","loc_ln":222,"loc_mod":"LaunchDarkly.Server.Network.Streaming","loc_pkg":"launchdarkly-server-sdk-4.0.0-8S8QnD9GDQ4PRlPelRf1P"},"msg":"\"starting new streaming connection\"","sev":"Debug"}
{"at":"2023-09-26T06:56:18.970990153Z","loc":{"loc_col":6,"loc_fn":"src/LaunchDarkly/Server/Network/Streaming.hs","loc_ln":193,"loc_mod":"LaunchDarkly.Server.Network.Streaming","loc_pkg":"launchdarkly-server-sdk-4.0.0-8S8QnD9GDQ4PRlPelRf1P"},"msg":"\"starting readStream\"","sev":"Error"}
```

**SDK version**
`4.0.0`

**Language version, developer tools**
GHC 9.4.4

**OS/platform**
NixOS

**Additional context**
N/A.

**Comment by keelerm84** [comment]
Thank you for raising this issue to our attention. I will take a look at this and let you know once a fix has been released.

**Comment by keelerm84** [comment]
@sir4ur0n sorry for the delay on this! This has been addressed in v4.0.1.
